### ISSUE TITLE
3 Channel RGB Models not Displaying Properly

### ISSUE THREAD

**Issue by dcwehw99** [smeighan/xLights#525]
2016.18
Starting with 2016.15, i.e. the major changes to the layout tab, my 3 channel RGB models have not been displaying properly in the layout tab and in playing the sequence.  
I have six (6) windows with each window having a single-line 3 channel RGB string type for the top, bottom, left and right, i.e. I have four (4) separate 3 channel controllers per window.  
In the layout tab all my 3 channel RGB strings for the windows are missing, i.e. they no longer show around the windows, i.e. it's like they are now completely transparent.  Over on the extreme right of the picture are four (4) horizontal lines that appear to be the tops and bottoms of two of the six (6) windows.  The windows are located in the left-center of the picture.  
If you click around the window frames you can eventually find a model when the end points of the line turn on.  You can move it around but once you click off of it it disappears again.  If I change the string properties from 3 channel RGB to say RGB node everything appears in the correct locations and shows up when I play the sequence.  
When I play the sequence nothing shows around the windows any longer but those four (4) horizontal lines change colors like the tops and bottoms of the windows use to.  All the models are still located around the windows but they just don't display in the correct position or at all.

Denny

**Comment by AzGilrock** [comment]
I tried 3 channel RGB models and they look fine to me.  If you switch your View to "All Models" are all the checkboxes checked?

**Comment by dcwehw99** [comment]
All the check boxes are checked if I switch to the "All Models" view.  If I add new 3 channel RGB models they show up correctly in both the layout and sequence tabs.  It's only 3 channel RGB models that existed prior to 2016.15 that seem to be affected.  If I go back to release 2016.14 everything displays perfectly in the layout and sequence tabs.  I'm using the exact same Xlights rgbeffects XML file for both releases (14 & 18).  Was there some type of conversion done to the models in going from 2016.14 to 2016.15?

If I take one of the 3 channel RGB models and only change the "string type" to RGB Nodes it then shows up correctly in the layout and sequence tabs.  If I then change it back to 3 channel RGB it's gone.  I do have other models that are "3 channel RGB" string type that work correctly but they are not "Single-Line" models.  It appears to be the combination of "Single-Line" and "3 Channel RGB" that existed prior to release 2016.15 that doesn't work. BTW any of the "xxx Nodes" string types work.

Denny

**Comment by dcwehw99** [comment]
I just figured out what's causing the models to disappear.  
I compared a new model I created to my existing ones and see that the only difference between them for the "Single Line" model is the field "Starting Location".  My existing models are defined as "Right" and the new one I created is defined as "Left".  If I change mine from "Right" to "Left" they show up in the layout and sequence tabs properly.  If I change the new one to "Right" it disappears.

It appears to be an issue with using "Right" in the "Starting Location" field.

Denny

**Comment by AzGilrock** [comment]
I still can't cause the problem so it's unlikely to be fixed unless you provide the files.

**Comment by dcwehw99** [comment]
I've attached a video of the problem that I've been able to recreate on three different computers.  It's pretty much self-explanatory.  I started Xlights 2016.18 without any models defined so as to not muddy the water.
1.  I created a "Single Line" model with all the defaults.
2.  I changed the model name to "Single Line - Left".
3.  I changed the "String Type" from the default of "RGB Nodes" to "3 Channel RGB".
4.  I then created a second "Single Line" model with all the defaults.
5.  I changed the model name to "Single Line - Right".
6.  I changed the "String Type" from the default of "RGB Nodes" to "3 Channel RGB".
7.  I then changed the "Starting Location" from the default of "Left" to "Right" and the model then disappears.

If you use the model name "Single Line - Right" in a sequence it doesn't ever show up just like it doesn't show in the layout tab.

Denny
[Xlights Single-Line Right Bug.zip](https://github.com/smeighan/xLights/files/211965/Xlights.Single-Line.Right.Bug.zip)
